src/cmd/main.cpp

```diff
@@ -44,58 +52,77 @@ int main(int argc, const char* argv[])
              << endl;
         cout << "[USAGE] " << argv[0]
              << " <32-byte private_key> <33-byte public_key> "
-                "<listen_ip_address> <listen_port> <1 if loadConfig, 0 "
+                "<listen_ip_address or NAT> <listen_port> <1 if loadConfig, 0 "
                 "otherwise> <SyncType, 0 for no, 1 for new,"
                 " 2 for normal, 3 for ds, 4 for lookup> "
                 "<1 if recovery, 0 otherwise>"
              << endl;
+        return 0;
     }
-    else
+    else if (string(argv[3]) == "NAT")
     {
-        INIT_FILE_LOGGER("zilliqa");
-        INIT_STATE_LOGGER("state");
-        INIT_EPOCHINFO_LOGGER("epochinfo");
-
-        vector<unsigned char> tmpprivkey
-            = DataConversion::HexStrToUint8Vec(argv[1]);
-        vector<unsigned char> tmppubkey
-            = DataConversion::HexStrToUint8Vec(argv[2]);
-        // PrivKey privkey(tmpprivkey, 0);
-        PrivKey privkey;
-        if (privkey.Deserialize(tmpprivkey, 0) != 0)
+        NAT nt;
```

Comment:
`nt` is local within the `if` which means that by the end of the `if` scope it will be destructed and `removeRedirect()` will be called. To make this work, probably use a `std::unique_ptr` to keep it alive till end of `main()` if the pointer is initialized.